CreeDictionary/API/templatetags/inflection_extras.py

```diff
@@ -77,3 +79,36 @@ def presentational_pos(wordform: Union[Wordform, dict]) -> str:
         f"can not determine presentational pos for {wordform_dict}, id={wordform_dict['id']}"
     )
     return ""
+
+
+# XXX: mypy can't infer this type?
+PER_REQUEST_ID_COUNTER = WeakKeyDictionary()  # type: WeakKeyDictionary
```

Comment:
does `WeakKeyDictionary` help with memory leak because the keys are garbaged collected as soon as the reference counter to the dictionary is 0?? Want some explanation.

Reply:
Correct! When the `Response` object reaches a reference count of 0, the key in the dictionary is invalid, and thus, its data can be deleted